packages/eas-cli/src/analytics.ts

```diff
@@ -0,0 +1,82 @@
+import Segment from 'analytics-node';
+import os from 'os';
+
+/**
+ * We use require() to exclude package.json from TypeScript's analysis since it lives outside
+ * the src directory and would change the directory structure of the emitted files
+ * under the build directory
+ */
+const packageJSON = require('../package.json');
+
+const PLATFORM_TO_ANALYTICS_PLATFORM: { [platform: string]: string } = {
+  darwin: 'Mac',
+  win32: 'Windows',
+  linux: 'Linux',
+};
+
+// values shared between different instances of Segement
+let sharedUsername: string | undefined;
+let sharedUserTraits: any;
+
+export class AnalyticsClient<Event extends string> {
+  private instance?: Segment;
+  private userIdentifyCalled = false;
+
+  constructor(writeKey?: string) {
+    this.instance = writeKey ? new Segment(writeKey) : undefined;
+  }
+
+  public setUserProperties(username: string, traits: any) {
+    sharedUsername = username;
+    sharedUserTraits = traits;
+
+    this.ensureUserIdentified();
+  }
+
+  public logEvent(name: string, properties: any = {}) {
+    if (this.instance && sharedUsername) {
```

Comment:
```suggestion
    if (this.instance) {
```
Maybe it should be possible to also log an event without being signed in?

Reply:
Currently, there are no events that would need that and `eas-cli` for almost everything will require API interaction